### PR TITLE
Gstreamer cleanup

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

Paving the road towards switching to gstreamer 1.8 as the default
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by zimbatm** [comment]
Everything is compiling except xulrunner (and gnash because xulrunner is a dep), but the issue is unrelated to these changes.

**Comment by zimbatm** [comment]
I am thinking of re-introducing a top-level gstreamer in the form of a configurable buildEnv that includes the various plugins, what do you think?

**Comment by vcunat** [comment]
That probably won't help, as the plugins tend to be not found unless variables are set during runtime, IIRC. I think `wrapGAppsHook` can already do that automatically, based on your build inputs.

**Comment by zimbatm** [comment]
Yeah I had to wrap that. Right now I'm having the following issue when I start my shell:
```
Package gstreamer-1.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gstreamer-1.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gstreamer-1.0' found
bash: /gst: No such file or directory
```

File: pkgs/applications/science/robotics/qgroundcontrol/default.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
   , qtbase, qtlocation, qtserialport, qtdeclarative, qtconnectivity, qtxmlpatterns
   , qtsvg, qtquick1, qtquickcontrols, qtgraphicaleffects, qmakeHook
   , makeQtWrapper, lndir
-  , gst_all_1, qt_gstreamer1, pkgconfig, glibc
+  , gst_all_1, qt-gstreamer1, pkgconfig, glibc
```
Comment (rasendubi) [inline pkgs/applications/science/robotics/qgroundcontrol/default.nix]: `qt-gstreamer1` is not used here.

File: pkgs/applications/video/gnash/default.nix [hunk 2]
```diff
@@ -71,10 +70,10 @@ stdenv.mkDerivation rec {
          --without-gconf
          --enable-gui=gtk"
 
-       # In `libmedia', Gnash compiles with "-I$gst_plugins_base/include",
-       # whereas it really needs "-I$gst_plugins_base/include/gstreamer-0.10".
+       # In `libmedia', Gnash compiles with "-I$gst-plugins-base/include",
```
Comment (rasendubi) [inline pkgs/applications/video/gnash/default.nix]: Replace with `${...}` as the variable name contains hyphens. (I know, that's just a comment but still confusing.)

Comment (vcunat) [inline pkgs/applications/video/gnash/default.nix]: IIRC bash variables can't contain dashes in their name; ref: http://stackoverflow.com/a/2821201/587396

Comment (zimbatm) [inline pkgs/applications/video/gnash/default.nix]: Luckily it's a nix variable which we want to get the derivation path from. For a moment I thought this PR was going to be closed :sweat:

Comment (zimbatm) [inline pkgs/applications/video/gnash/default.nix]: Fixed in de6b23568be81c0fd60e09917e6a2af54e5e8545

File: pkgs/applications/science/robotics/qgroundcontrol/default.nix (hunk 1 above)
Comment (zimbatm) [inline pkgs/applications/science/robotics/qgroundcontrol/default.nix]: Fixed in e6d8405cce313f65cd3cbac1c367b0b24ef2b565

**Pull request closed without merging**